### PR TITLE
Fixed incorrect asserts

### PR DESCRIPTION
**Pull request by rendel**
It is not possible to use a range distribution with the current assert, it will always fail.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/RelDistributions.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ private RelDistributionImpl(Type type, ImmutableIntList keys) {
           || Ordering.natural().isOrdered(keys)
           : "key columns of hash distribution must be in order";
       assert type == Type.HASH_DISTRIBUTED
-          || type == Type.RANDOM_DISTRIBUTED
+          || type == Type.RANGE_DISTRIBUTED
           || keys.isEmpty();
     }
```
Comment (chunweilei) [inline core/src/main/java/org/apache/calcite/rel/RelDistributions.java:108]: @rendel Thank you for reporting this. I think it is a bug. Could you file a JIRA for this?

**Comment by chunweilei** [comment]
BTW, you can find more details about contribution in the doc[1].

[1] https://calcite.apache.org/develop/#contributing

File: core/src/main/java/org/apache/calcite/rel/RelDistributions.java (hunk 1 above)
Comment (vlsi) [reply core/src/main/java/org/apache/calcite/rel/RelDistributions.java:108]: I would say a test case is more important than JIRA.
If the assertion is user-facing, then it should be a regular if + illegalargumentexception

**Comment by chunweilei** [comment]
Let us move to https://github.com/apache/calcite/pull/2822.

**Pull request closed without merging**